carousel/core/models.py

```diff
@@ -218,7 +218,16 @@ def _initialize(self):
             if src_value:
                 value = src_model[layer] = src_value
             else:
-                value = dict(value['sources'])
+                srcmod, srcpkg = value.get('module'), value.get('package')
+                try:
+                    value = dict(value['sources'])
+                except ValueError:
+                    value = dict.fromkeys(value['sources'], {})
+                for src in value.viewkeys():
+                    if srcmod is not None:
+                        value[src]['module'] = srcmod
+                    if srcpkg is not None:
+                        value[src]['package'] = srcpkg
             # set layer attribute with model data
             setattr(self, layer, layer_cls(value))
```

Comment:
Yes, I completely, agree, I hated this section, and it was a pain to write, but this is the way that models are being imported right now, since they were originally JSON, so they get deserialized back to JSON and re-imported. There is a [fast import method in formulas](../blob/953a7d8725e30f39e8ed3ed002a1c657eac4d6f9/carousel/core/formulas.py#L99-L102), that I should probably add to layers, but I didn't so this is the ridiculous way it's being done now :frowning_face: